source/docs/software/hardware-apis/pneumatics/pneumatics.rst

```diff
@@ -6,89 +6,101 @@ Operating pneumatic cylinders
 
-        Compressor c = new Compressor(0, PneumaticsModuleType.CTREPCM);
+        Compressor pcmCompressor = new Compressor(0, PneumaticsModuleType.CTREPCM);
+        Compressor phCompressor = new Compressor(1, PneumaticsModuleType.REVPH);
 
-        c.setClosedLoopControl(true);
-        c.setClosedLoopControl(false);
+        pcmCompressor.enableDigital();
+        pcmCompressor.disable();
 
-        boolean enabled = c.enabled();
-        boolean pressureSwitch = c.getPressureSwitchValue();
-        double current = c.getCompressorCurrent();
+        boolean enabled = pcmCompressor.enabled();
+        boolean pressureSwitch = pcmCompressor.getPressureSwitchValue();
+        double current = pcmCompressor.getCompressorCurrent();
 
     .. code-tab:: c++
 
-        frc::Compressor c{0, frc::PneumaticsModuleType::CTREPCM};
+        frc::Compressor pcmCompressor{0, frc::PneumaticsModuleType::CTREPCM};
+        frc::Compressor phCompressor{1, frc::PneumaticsModuleType::REVPH};
 
-        c.SetClosedLoopControl(true);
-        c.SetClosedLoopControl(false);
-
-        bool enabled = c.Enabled();
-        bool pressureSwitch = c.GetPressureSwitchValue();
-        double current = c.GetCompressorCurrent();
+        pcmCompressor.EnableDigital();
+        pcmCompressor.Disable();
 
+        bool enabled = pcmCompressor.Enabled();
+        bool pressureSwitch = pcmCompressor.GetPressureSwitchValue();
+        double current = pcmCompressor.GetCompressorCurrent();
 
 
+The Pneumatic Hub also has methods for enabling compressor control using the REV Analog Pressure Sensor (``enableAnalog`` method).
 
 Solenoid control
 ----------------
 
-FRC teams use solenoids to perform a variety of tasks, from shifting gearboxes to operating robot mechanisms. A solenoid is a valve used to electronically switch a pressurized air line "on" or "off". For more information on solenoids, see `this wikipedia article <https://en.wikipedia.org/wiki/Solenoid_valve>`__. Solenoids are controlled by a robot's Pneumatics Control Module, or PCM, which is in turn connected to the robot's roboRIO via CAN. The easiest way to see a solenoid's state is via the small red LED (which indicates if the valve is "on" or not), and solenoids can be manually actuated when un-powered with the small button adjacent to the LED.
+FRC teams use solenoids to perform a variety of tasks, from shifting gearboxes to operating robot mechanisms. A solenoid is a valve used to electronically switch a pressurized air line "on" or "off". For more information on solenoids, see `this wikipedia article <https://en.wikipedia.org/wiki/Solenoid_valve>`__. Solenoids are controlled by a robot's Pneumatics Control Module, or Pneumatic Hub, which is in turn connected to the robot's roboRIO via CAN. The easiest way to see a solenoid's state is via the LEDs on the PCM or PH (which indicates if the valve is "on" or not), and solenoids can be manually actuated when un-powered with the small button adjacent to the LED.
```

Comment:
This still implies that the button is located on the PCM/PH, and not the solenoid.

Reply:
```suggestion
FRC teams use solenoids to perform a variety of tasks, from shifting gearboxes to operating robot mechanisms. A solenoid is a valve used to electronically switch a pressurized air line "on" or "off". For more information on solenoids, see `this wikipedia article <https://en.wikipedia.org/wiki/Solenoid_valve>`__. Solenoids are controlled by a robot's Pneumatics Control Module, or Pneumatic Hub, which is in turn connected to the robot's roboRIO via CAN. The easiest way to see a solenoid's state is via the LEDs on the PCM or PH (which indicates if the valve is "on" or not), and solenoids can be manually actuated when un-powered with the small button on the solenoid.
```